apps/ewallet/lib/ewallet/exporters/csv_exporter.ex

```diff
@@ -0,0 +1,94 @@
+defmodule EWallet.CSVExporter do
+  @moduledoc """
+  Entry point for a CSV exporter. Runs a GenServer that will take
+  care of the export using the appropriate adapter.
+  """
+  use GenServer
+  import Ecto.Query
+  alias EWallet.{S3Adapter, GCSAdapter, LocalAdapter, Exporter}
+  alias EWalletDB.{Repo, Export}
+  alias Utils.Helper.PidHelper
+
+  def start(export, schema, query, serializer) do
+    with count <- get_count(query),
+         {:ok, record_estimated_size} <- get_size_estimate(query, serializer),
+         estimated_size <- record_estimated_size * count,
+         {:ok, export} <- Export.init(export, schema, count, estimated_size, %Exporter{}),
+         {:ok, pid} <-
+           GenServer.start_link(
+             __MODULE__,
+             [
+               export: export,
+               query: query,
+               serializer: serializer
+             ],
+             name: {:global, export.uuid}
+           ),
+         {:ok, export} <-
+           Export.update(export, %{
+             pid: PidHelper.pid_to_binary(pid),
+             originator: %Exporter{}
+           }),
+         :ok <- GenServer.cast(pid, :upload) do
+      {:ok, pid, export}
```

Comment:
What's the pid used for?

Reply:
The PID is used to identify the process responsible for exporting. It is returned to the client and could potentially be used to check what's up with a process. Right now, it's mostly used in order to have functional tests :D